base/applications/cmdutils/tasklist/tasklist.c

```diff
@@ -0,0 +1,359 @@
+    NTSTATUS Status = NtQuerySystemInformation(SystemProcessInformation, NULL, 0, &ResultLength);
+
+    // New process/thread might appear before we call for the actual data.
+    // Try to avoid this by retrying several times.
+    for (INT Retry = 0; Retry < NT_SYSTEM_QUERY_MAX_RETRY; Retry++)
+    {
+        // (Re)allocate buffer
+        ProcessInfoBufferLength = ResultLength;
+        ResultLength = 0;
+        if (ProcessInfoBuffer)
+        {
+            PBYTE NewProcessInfoBuffer = HeapReAlloc(GetProcessHeap(), 0,
+                                                     ProcessInfoBuffer,
+                                                     ProcessInfoBufferLength);
+            if (NewProcessInfoBuffer)
+            {
+                ProcessInfoBuffer = NewProcessInfoBuffer;
+            }
+            else
+            {
+                // out of memory ?
+                ConResMsgPrintf(StdOut, 0, IDS_OUT_OF_MEMORY);
+                HeapFree(GetProcessHeap(), 0, ProcessInfoBuffer);
+                return FALSE;
+            }
+        }
+        else
+        {
+            ProcessInfoBuffer = HeapAlloc(GetProcessHeap(), 0, ProcessInfoBufferLength);
+            if (!ProcessInfoBuffer)
+            {
+                // out of memory ?
+                ConResMsgPrintf(StdOut, 0, IDS_OUT_OF_MEMORY);
+                return FALSE;
+            }
+        }
+
+        // Query information
+        Status = NtQuerySystemInformation(SystemProcessInformation,
+                                          ProcessInfoBuffer,
+                                          ProcessInfoBufferLength,
+                                          &ResultLength);
+        if (Status != STATUS_INFO_LENGTH_MISMATCH)
+        {
+            break;
+        }
+    }
+
+    if (!NT_SUCCESS(Status))
+    {
+        // tried NT_SYSTEM_QUERY_MAX_RETRY times, or failed with some other reason
+        ConResMsgPrintf(StdOut, 0, IDS_ENUM_FAILED);
+        HeapFree(GetProcessHeap(), 0, ProcessInfoBuffer);
+        return FALSE;
+    }
+
+    ConPrintf(StdOut, L"\n");
+
+    if (!bNoHeader)
+    {
+        PrintHeader();
+    }
+
+    PSYSTEM_PROCESS_INFORMATION pSPI;
+    pSPI = (PSYSTEM_PROCESS_INFORMATION)ProcessInfoBuffer;
+    while (pSPI)
+    {
+        PrintString(pSPI->UniqueProcessId ? pSPI->ImageName.Buffer : L"System Idle Process", COLUMNWIDTH_IMAGENAME, TRUE);
+        PrintSpace(1);
+        PrintNum((LONGLONG)(INT_PTR)pSPI->UniqueProcessId, COLUMNWIDTH_PID);
+        PrintSpace(1);
+        PrintNum((ULONGLONG)pSPI->SessionId, COLUMNWIDTH_SESSION);
+        PrintSpace(1);
+        PrintMemory(pSPI->WorkingSetSize, COLUMNWIDTH_MEMUSAGE, 1);
+
+        ConPrintf(StdOut, L"\n");
+
+        if (pSPI->NextEntryOffset == 0)
+            break;
+        pSPI = (PSYSTEM_PROCESS_INFORMATION)((LPBYTE)pSPI + pSPI->NextEntryOffset);
+    }
+
+    HeapFree(GetProcessHeap(), 0, ProcessInfoBuffer);
+    return TRUE;
+}
+
+INT GetArgumentType(WCHAR* argument)
```

Comment:
```suggestion
INT GetArgumentType(LPCWSTR argument)
```

Reply:
or PCWSTR if the rest of the code tends to use the latter, or its other versions: PWSTR, PWCHAR etc.